### PR TITLE
typescript-view-swapping

### PR DESCRIPTION
**Pull request by TristanChao**
Screenshot typescript view swapping

### CODE REVIEW

File: typescript-view-swapping/main.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    $allViews[i].className = 'view hidden';
+  }
+
+  const currentView = $eventTarget.dataset.view;
+
+  for (let i = 0; i < $allViews.length; i++) {
+    if ($allViews[i].getAttribute('data-view') === currentView) {
+      $allViews[i].className = 'view';
+    }
+  }
```
Comment (brandon-moy) [inline typescript-view-swapping/main.ts:35]: You should try to combine these two `for` loops into one `for` loop. You are looking over the same item, in the same direction. Try to make it more efficient!

**Review by brandon-moy — CHANGES_REQUESTED** [review]
Great job! One thing to change here.

File: typescript-view-swapping/main.ts (hunk 1 above)
Comment (TristanChao) [reply typescript-view-swapping/main.ts:35]: I was trying to ensure that one view wouldn't appear before the current view was hidden. Is that not really something to be concerned about here?

**Review by brandon-moy — APPROVED** [review]
👍

**Pull request merged**